### PR TITLE
[Concept Entry] Python Deque Deque #4452

### LINKED ISSUE #4452: [Concept Entry] Python Deque Deque
### Reason/inspiration (optional)

"We would like a new concept entry for [Python](https://www.codecademy.com/resources/docs/python) on Deque.
Provide the entry in a markdown file under `docs/content/python/concepts/deque/deque.md`

The entry should
    - Have an introductory paragraph about the deque data structure.
    - A syntax section with a pseudo code block with an explanation of it.
    - An example showing the use of deque.
    - Explain the equivalent method for stacks and queue.
Please refer to the [concept entry templates](https://github.com/Codecademy/docs/blob/main/documentation/concept-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue."

### Entry Type (select all that apply)

New Concept Entrty for Existing Topic

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by Aquacoder345**
### Description

- Added the Concept Entry Python Deque

### Issue Solved

- Closes #4452 

### Type of Change

- Adding a new entry

### Checklist

- [x ] All writings are my own.
- [x ] My entry follows the Codecademy Docs style guide.
- [x ] My changes generate no new warnings.
- [x ] I have performed a self-review of my own writing and code.
- [x ] I have checked my entry and corrected any misspellings.
- [x ] I have made corresponding changes to the documentation if needed.
- [ x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [ x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [ x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

File: content/python/concepts/deque/deque.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+## Remove and return the leftmost item
+
+item = d.popleft()
+
+## Rotate the deque n steps to the right
+
+d.rotate(n)
+
+In this syntax, deque() creates a new deque object, which can be initialized with an optional iterable and a maximum length. The append() and appendleft() methods add items to the right and left ends, respectively.
+The pop() and popleft() methods remove items from the corresponding ends. The rotate() method moves elements from one end to the other, effectively rotating the deque.
```
Comment (letsbuildanything) [inline content/python/concepts/deque/deque.md:44]:
```suggestion
```

Comment (letsbuildanything) [reply content/python/concepts/deque/deque.md:44]: The description of these methods is not required here. In the future, they will be added as term entries.

File: content/python/concepts/deque/deque.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+## Using deque
```
Comment (letsbuildanything) [inline content/python/concepts/deque/deque.md:46]:
```suggestion
## Example
```

File: content/python/concepts/deque/deque.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+browser.forward() # Outputs: Forward to: about.html
+
+```Popping
+from the left
+left_item = d.popleft()  # returns 0, deque becomes [1, 2, 3]
+```
+
+In this example, we have a BrowserHistory class that uses two deques: one for the main history and another as a stack for the forward history. When a new page is visited, it is appended to the history deque, and the forward history is cleared.
+
+The back() method pops from the history and pushes onto the forward stack, while the forward() method does the opposite.
```
Comment (letsbuildanything) [inline content/python/concepts/deque/deque.md:94]: Below are the following changes or suggestion to be made here:
- explain use of `deque` object with easy to understand and beginner-friendly code. for example, using simple array of numbers or objects.
- Write Concrete code. for example in your case, you haven't mentioned `import` statement. (`from collections import deque`)
- Write all the runnable code inside triple backticks (```).
- For output, use ` ```shell` not ` ```Popping`. To learn more about this, you can visit [Style guide](https://github.com/Codecademy/docs/blob/main/documentation/style-guide.md)

**Review by letsbuildanything — CHANGES_REQUESTED** [review]
@Aquacoder345 I've requested some changes below. Please make them ASAP.

File: content/python/concepts/deque/deque.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+## Equivalent Methods for Stacks and Queues
+
+- Stacks: To implement a stack using a deque, you would use the append() method to push items onto the stack and the pop() method to remove them.
+- Queues: To implement a queue, you would use the append() method to enqueue items at the back and the popleft() method to dequeue items from the front.
```
Comment (letsbuildanything) [inline content/python/concepts/deque/deque.md:99]: - use backticks i.e. \`append()\`,  for methods, properties, variables, etc. 
- Avoid using pronouns like `you`, `we`, etc
- Lastly, make all the necessary changes you deem appropriate here.

**Review by letsbuildanything — CHANGES_REQUESTED** [review]

**Comment by letsbuildanything** [comment]
@Aquacoder345 
you've committed changes to the `package.json` and `yarn.lock` files. Please remove these changes.

**Comment by Aquacoder345** [comment]
@letsbuildanything I believe I found and delete the changes made to the package.json. To my knowledge I did not make any manual changes to the yarn.lock file, are you able to assist me with finding what changes I made to yarn.lock.

**Comment by letsbuildanything** [comment]
Hey @Aquacoder345,
I’ve reverted the changes in the `yarn.lock` file. Could you please update your contribution file to ensure it aligns with Codecademy's contribution guidelines? Additionally, make sure to review and follow our [style guide](https://github.com/Codecademy/docs/blob/main/documentation/style-guide.md).

Thank you!

**Comment by Aquacoder345** [comment]
Thank you @letsbuildanything. I will get this resolved as soon as i possible can.

**Comment by letsbuildanything** [comment]
@Aquacoder345 It's been a week since the request was made. Please make the necessary changes as soon as possible.
Thank you!

**Comment by Aquacoder345** [comment]
@letsbuildanything I am in the process of moving and will get these changes made. apology in getting these changes completed.

**Comment by letsbuildanything** [comment]
@Aquacoder345, please ensure you close an existing PR before opening a new one for the same issue. 
since you've opened a similar new [PR#4759](https://github.com/Codecademy/docs/pull/4759), I am closing this PR.

**Pull request closed without merging**